joplin/base/views/new_page_from_modal.py

```diff
@@ -12,17 +12,31 @@
 from pages.information_page.models import InformationPage
 from pages.topic_page.models import TopicPage
 from pages.topic_collection_page.models import TopicCollectionPage
+from pages.topic_collection_page.factories import create_topic_collection_page_from_page_dictionary
 from pages.department_page.models import DepartmentPage
 from pages.official_documents_page.models import OfficialDocumentPage
 from pages.guide_page.models import GuidePage
 from pages.form_container.models import FormContainer
 from pages.location_page.models import LocationPage
 from pages.event_page.models import EventPage
 from groups.models import Department
+from importer.page_importer import PageImporter
 from base.models.site_settings import JanisBranchSettings
 from django.contrib.contenttypes.models import ContentType
 import json
 
+
+def import_page_from_url(url):
+    page_importer = PageImporter(url)
+
+    revision_id = page_importer.revision_id
+    page_dictionary = page_importer.get_page_dictionary_from_revision()
+
+    page = page_importer.create_page(page_dictionary, revision_id)
```

Comment:
page_importer shouldn't need you to plug in your page_dictionary and revision_id back in, right? Aren't they already saved inside the instance?

Public methods on importer:
page_importer.get_page_dictionary_from_revision() - for the sake of testing/verification
page_importer.create_page() - creating the actual page.

Creating a page could even be a 1 liner:
page = PageImporter(preview_url).create_page()

getting the page_dictionary could be even part of the __init__ process.

Reply:
I was debating where the page_dictionary should live, I like the idea of it living in the page_importer now that I've gone the "each page has everything it needs in one query" route

I kind of like that the init for importer is just initializing it's ability to import, but a one liner could be:
 PageImporter(preview_url).get_page().create_page()
by updating get_page_dictionary_from_revision() to save to the importer itself instead of returning it out